### DISCUSSION TITLE
Comparação de ViewModels X DTOs

### DISCUSSION THREAD

**Discussion by wmnogueira** [balta-io/2811#33] [Q&A] [answered]
Bom dia a todo(a)s, Balta navegando na internet encontrei alguns artigos e explicações de DTOs, consegui chegar a conclusão de que são tipo Views que criamos no SQL Server melhoradas. Agora seguindo o curso Fundamentos AspNet vi que temos as classes em ViewModels. Pergunta:  Posso interpretar que As classes ViewsModels é  mesma coisa que as classes  DTOS??

Ja peço desculpa se dei uma "Viajada", rss

Marcelinho

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde @wmnogueira, como vai?

Na verdade não necessariamente...
DTO é a sigla para Data Transfer Object ou objeto de transferência de dados. Normalmente são objetos simples (Anêmicos) sem comportamento, visto que seu objetivo é transportar dados.

Então podemos usá-los inclusive para Views do SQL Server, mas não precisa ser só para isto... podemos transferir informações de uma camada para outra com DTOs..

Uma ViewModel é a representação de dados de uma View (Inclusive indo a fundo seria errado chamar algo de ViewModel nas APIs, mas é comum esta nomenclatura).

Os ViewModels são uma espécie de DTO, mas podem conter estado da página, comandos, ações e diversos outros itens... é como se fossem uma extensão...

Tem muito mais coisas sobre as View Models, você pode procurar por MVVM para entender melhor <3
